rpc/api.py

```diff
@@ -509,16 +510,184 @@ class CapabilityViewSet(viewsets.ReadOnlyModelViewSet):
     serializer_class = CapabilitySerializer
 
 
-class ClusterViewSet(viewsets.ReadOnlyModelViewSet):
-    # todo: handle create/update operations and change to viewsets.ModelViewSet
+class ClusterViewSet(viewsets.ModelViewSet):
     queryset = Cluster.objects.all()
     serializer_class = ClusterSerializer
     lookup_field = "number"
+    http_method_names = ["get", "post", "patch", "head", "options"]
 
     def get_queryset(self):
         """Get clusters with RFC number annotations"""
         return Cluster.objects.with_rfc_number_annotated()
 
+    @extend_schema(
+        operation_id="clusters_add_document",
+        request=inline_serializer(
+            name="ClusterAddDocument",
+            fields={
+                "draft_name": serializers.CharField(
+                    help_text="Name of the draft to add to the cluster"
+                ),
+            },
+        ),
+        responses=ClusterSerializer,
+        examples=[
+            OpenApiExample(
+                "Add Document to Cluster",
+                value={"draft_name": "draft-ietf-example-document"},
+                request_only=True,
+            ),
+        ],
+    )
+    @action(detail=True, methods=["post"], url_path="add-document")
+    def add_document(self, request, number=None):
+        """Add a document to a cluster"""
+        cluster = self.get_object()
+        draft_name = request.data.get("draft_name")
+
+        if not draft_name:
+            return Response(
+                {"error": "draft_name is required"},
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        # Get the Document by draft name
+        try:
+            doc = Document.objects.get(name=draft_name)
+        except Document.DoesNotExist:
+            return Response(
+                {"error": f"Document with name '{draft_name}' not found"},
+                status=status.HTTP_404_NOT_FOUND,
+            )
+
+        # Check if document is already in any cluster
+        existing_member = ClusterMember.objects.filter(doc=doc).first()
+        if existing_member:
+            return Response(
```

Comment:
See above re: raising exceptions. If you squint this is still a validation error